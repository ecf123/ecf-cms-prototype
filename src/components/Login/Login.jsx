import "./Login.scss";
import { useState } from "react";
import Button from "../Button/Button";
import LoginInputBox from "../LoginInputBox/LoginInputBox";
import { useNavigate } from "react-router-dom";
import { useEffect } from "react";

const Login = ({ setCredential, setLogIn }) => {
  const [hasSubmitted, setHasSubmitted] = useState(false);
  const [hasError, setHasError] = useState(false);

  const handleSubmit = (e) => {
    e.preventDefault();

    const email = e.target.elements[0].value;
    const password = e.target.elements[1].value;

    if (!email || !password) {
      setHasError(true);
      return;
    }

    setHasError(false);
    setLogIn(true);

    const user = {
      email: email,
      password: password,
    };
    setCredential(user);
    setHasSubmitted(true);
  };

  const navigate = useNavigate();
  useEffect(() => {
    navigate("/");
  }, [hasSubmitted]);

  return (
    <>
      {!hasError && hasSubmitted && (
        <div>
          <p data-testid="success" className="login__success">
            Thank you for submitting! We will be in touch
          </p>
        </div>
      )}
      {!hasSubmitted && (
        <form
          data-testid="form"
          onSubmit={handleSubmit}
          name="form"
          className="login"
        >
          <div className="login__title">
            <h1 className="login__title--heading">Welcome Back</h1>
            <p className="login__title--text">
              Log into your account to access the CMS for the ECF Future Hub App.
            </p>
          </div>
          <div>
            <LoginInputBox
              labelText="Email Address"
              inputType="email"
              labelId="email"
            />
            <LoginInputBox
              labelText="Password"
              inputType="password"
              labelId="password"
            />
            {hasError && (
              <div>
                <p className="login__alert">Sorry something went wrong </p>
              </div>
            )}
          </div>

          <Button
            style={"button login-black square-border"}
            textStyle={"text medium-text black"}
            buttonText={"Login"}
          />
        </form>
      )}
    </>
  );
};

export default Login;
