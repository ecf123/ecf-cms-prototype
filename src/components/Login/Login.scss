@use "../../styles/variables.scss" as *;
@use "../../styles/typography.scss" as *;

    .alert{
        color: red;
    }

    label {
        font-family: $font-primary;
        color: $color-black;
        font-size: 0.75rem;
        font-weight: 500;
        line-height: 1.2rem;
    }

    #password-input, #email-input{
        height: 2.813rem;
        width: 22.125rem;
        border: 0.063rem solid #E8E8E8;
        border-radius: 0.25rem;
        color: #000000;
        margin-top: 0.313rem;
        margin-bottom: 0.625rem;
        
    }


    #form-submit{
        background-color: $color-green;
        width: 22.125rem;
        height: 2.875rem;
        border-radius: 0.25rem;
        gap: 0.625rem;
        font-size: 0.75rem;
        font-weight: 500;
        line-height: 1.2rem;
        margin-top: 1.875rem;
    }

    .success{
        color:#000000;
    }

    
