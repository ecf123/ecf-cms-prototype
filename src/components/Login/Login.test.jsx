import { render, screen, fireEvent } from "@testing-library/react";
import Login from "./Login";
import userEvent from "@testing-library/user-event";
import { customRender } from "../../utils/testUtils";

it("should render the form", () => {
  const { container } = customRender(<Login/>);
  expect(container).toMatchSnapshot();
});

it("should not render the error message on load", () => {
  render(<Login />);
  const errorMessage = screen.queryByText("Sorry something went wrong");

  expect(errorMessage).toBeFalsy();
});

it("shouldn't render the success message on load", () => {
  render(<Login />);
  const successMessage = screen.queryByText(
    "Thank you for submitting! We will be in touch"
  );
  expect(successMessage).toBeFalsy();
});

it("should not submit the form when not all credentials are valid and display error message", () => {
  render(<Login />);

  const emailInput = screen.getByLabelText("Email Address");
  userEvent.type(emailInput, "cheniangam63.com");

  const passwordInput = screen.getByLabelText("Password");
  userEvent.type(passwordInput, "fix");

  const button = screen.queryByText("Login");
  userEvent.click(button);

  const errorMessage = screen.getByText("Sorry something went wrong");
  const successMessage = screen.queryByText(
    "Thank you for submitting! We will be in touch"
  );

  expect(errorMessage).toBeTruthy();
  expect(successMessage).toBeFalsy();
});

it("should submit the form when all credentials are valid and display success message", () => {
  render(<Login />);

  const emailInput = screen.getByLabelText("Email Address");
  userEvent.type(emailInput, "lucy@gmail.com");

  const passwordInput = screen.getByLabelText("Password");
  userEvent.type(passwordInput, "nology2022");

  fireEvent.submit(screen.getByTestId("form"));

  const successMessage = screen.getByTestId("success");
  const errorMessage = screen.queryByText("Sorry something went wrong");

  expect(successMessage).toBeInTheDocument();
  expect(errorMessage).toBeFalsy();
});
