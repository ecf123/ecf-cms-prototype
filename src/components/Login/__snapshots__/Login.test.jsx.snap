// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the form 1`] = `
<div>
  <form
    class="login"
    data-testid="form"
    name="form"
  >
    <div
      class="login__title"
    >
      <h1
        class="login__title--heading"
      >
        Welcome Back
      </h1>
      <p
        class="login__title--text"
      >
        Feugiat magna viverra pellentesque habitant. Libero id amet nisl, tellus cras eleifend et in.
      </p>
    </div>
    <div>
      <div
        class="input-box"
      >
        <label
          class="input-box__label"
          for="Email Address"
        >
          Email Address
        </label>
        <input
          alt="input box"
          class="input-box__input short"
          id="Email Address"
          type="text"
        />
      </div>
      <div
        class="input-box"
      >
        <label
          class="input-box__label"
          for="Password"
        >
          Password
        </label>
        <input
          alt="input box"
          class="input-box__input short"
          id="Password"
          type="text"
        />
      </div>
    </div>
    <button
      class="button login-black square-border"
    >
      <div />
      <p
        class="text medium-text black"
      >
        Login
      </p>
    </button>
  </form>
</div>
`;
